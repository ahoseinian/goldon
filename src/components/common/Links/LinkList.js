import React from 'react';
import SingleLink from './SingleLink'

const LinkList = ({links}) => links.length
  ? (
    <div>
      {links.map(x => <SingleLink {...x} key={x._id}/>)}
    </div>
  )
  : null;

LinkList.propTypes = {
  links: React.PropTypes.arrayOf(React.PropTypes.object)
};

export default LinkList;
