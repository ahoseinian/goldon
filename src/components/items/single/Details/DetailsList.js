import React from 'react';
import SingleRowTable from '../../../common/SingleRowTable'
import LinkList from '../../../common/Links/LinkList'

const DetailsList = ({modelsNames, colors, names, links}) => (
  <div>
    <SingleRowTable name="مدل">{modelsNames.join(' , ')}</SingleRowTable>
    <SingleRowTable name="رنگهای موجود">{colors.join(' , ')}</SingleRowTable>
    <SingleRowTable name="نام های دیگر">{names.join(' , ')}</SingleRowTable>
    { links.length !== 0 &&
      <SingleRowTable name="لینک ها">
         <LinkList links={links}/>
      </SingleRowTable>
    }
  </div>
);

DetailsList.propTypes = {
  modelsNames: React.PropTypes.array.isRequired,
  colors: React.PropTypes.array.isRequired,
  names: React.PropTypes.array.isRequired,
  links: React.PropTypes.array
};

export default DetailsList;
