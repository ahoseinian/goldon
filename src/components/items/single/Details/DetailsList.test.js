import React from 'react'
import {shallow} from 'enzyme'
import DetailsList from './DetailsList'
import toJson from 'enzyme-to-json'
import LinkList from '../../../common/Links/LinkList'

const item = {
  modelsNames: [],
  colors: [],
  names: [],
  links: []
}

describe('DetailsList', () => {
  it('renders', () => {
    const wrapper = shallow(<DetailsList {...item}/>)
    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })

  it('does not render LinkList without links', () => {
    const wrapper = shallow(<DetailsList {...item}/>)
    expect(wrapper.find(LinkList).length).toEqual(0)
  })

  it('renders LinkList when there are links', () => {
    const itemWithLinks = {...item, links: [{}]}
    const wrapper = shallow(<DetailsList {...itemWithLinks}/>)
    expect(wrapper.find(LinkList).length).toEqual(1)
  })
})
