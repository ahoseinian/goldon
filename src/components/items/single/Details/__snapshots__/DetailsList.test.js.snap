exports[`DetailsList renders 1`] = `
<div>
  <SingleRowTable
    name="مدل" />
  <SingleRowTable
    name="رنگهای موجود" />
  <SingleRowTable
    name="نام های دیگر" />
</div>
`;
